packages/explat-client/src/create-explat-client.ts

```diff
@@ -0,0 +1,158 @@
+	 *
+	 * Call as many times as you like, it will only make one request at a time (per experiment) and
+	 * will only trigger a request when the assignment TTL is expired.
+	 *
+	 * Will never throw in production, it will return the default assignment.
+	 * It should not be run on the server but it won't crash anything.
+	 *
+	 * @param experimentName The experiment's name
+	 */
+	loadExperimentAssignment: ( experimentName: string ) => Promise< ExperimentAssignment >;
+
+	/**
+	 * Get an already loaded Experiment Assignment, will throw if there is an error, e.g. if it hasn't been loaded.
+	 *
+	 * Make sure loadExperimentAssignment has been called before calling this function.
+	 *
+	 */
+	dangerouslyGetExperimentAssignment: ( experimentName: string ) => ExperimentAssignment;
+}
+
+export class MissingExperimentAssignmentError extends Error {
+	constructor( ...params: any[] ) {
+		super( ...params );
+
+		// Maintains proper stack trace for where our error was thrown (only available on V8)
+		if ( Error.captureStackTrace ) {
+			Error.captureStackTrace( this, MissingExperimentAssignmentError );
+		}
+
+		this.name = 'MissingExperimentAssignmentError';
+	}
+}
+
+/**
+ * Create an ExPlat Client
+ *
+ * @param config Configuration object
+ */
+export default function createExPlatClient( config: Config ): ExPlatClient {
+	/**
+	 * The null experiment assignment we return when we can't retrieve one.
+	 */
+	const createNullExperimentAssignment = (): ExperimentAssignment => ( {
+		experimentName: 'null_experiment_assignment',
+		variationName: null,
+		retrievedTimestamp: 0,
+		ttl: 0,
+	} );
+
+	/**
+	 * This bit of code is the heavy lifting behind loadExperimentAssignment, allowing it to be used intuitively.
+	 *
+	 * AOAAT stands for Async One At A Time, this is how we ensure for each experiment that there is only ever one fetch process occuring.
+	 *
+	 * @param experimentName The experiment's name
+	 */
+	const createAOAATExperimentAssignmentFetchAndStore = ( experimentName: string ) =>
+		Timing.asyncOneAtATime( async () => {
+			const fetchedExperimentAssignment = await Request.fetchExperimentAssignment(
+				config,
+				experimentName
+			);
+			State.storeExperimentAssignment( fetchedExperimentAssignment );
+			return fetchedExperimentAssignment;
+		} );
+	const experimentNameToAOAATExperimentAssignmentFetchAndStore: Record<
+		string,
+		ReturnType< typeof createAOAATExperimentAssignmentFetchAndStore >
+	> = {};
+
+	return {
+		loadExperimentAssignment: async ( experimentName: string ): Promise< ExperimentAssignment > => {
+			try {
+				if ( ! Validation.isName( experimentName ) ) {
+					throw new Error( `Invalid experimentName: ${ experimentName }` );
+				}
+
+				const storedExperimentAssignment = State.retrieveExperimentAssignment( experimentName );
+				if (
+					storedExperimentAssignment &&
+					ExperimentAssignments.isAlive( storedExperimentAssignment )
+				) {
+					return storedExperimentAssignment;
+				}
```

Comment:
Yeah it is a good idea to add a safeguard around it.

A case where this could happen:
- `loadExperimentAssignment` called frequently AND
- No stored experiment or stored not alive AND
- Experiment fails to fetch quickly

Reply:
I have fixed this by storing the nullExperimentAssignment (now known as fallbackExperimentAssignment), with a 60s TTL.